vendor/csv_to_rkg.py

```diff
@@ -0,0 +1,186 @@
+import csv
+import binascii
+from dataclasses import dataclass
+
+
+# fmt: off
+class FaceButtonInput:
+    def __init__(self, AButton: int, BButton: int, ItemButton: int, csv_line: int):
+        assert AButton == 0 or AButton == 1, \
+            f"A button input has value `{AButton}` at line {csv_line}"
+
+        assert BButton == 0 or BButton == 1, \
+            f"B button input has value `{BButton}` at line {csv_line}"
+
+        assert ItemButton == 0 or ItemButton == 1, \
+            f"Item button input has value `{ItemButton}` at line {csv_line}"
+
+        self.AButton = AButton
+        self.BButton = BButton
+        self.ItemButton = ItemButton
+
+    def __eq__(self, other):
+        return (
+            self.AButton == other.AButton
+            and self.BButton == other.BButton
+            and self.ItemButton == other.ItemButton
+        )
+
+
+class DirectionInput:
+    def __init__(self, YDirection: int, XDirection: int, csv_line: int):
+        assert -7 <= YDirection <= 7, \
+            f"Y direction input has value `{YDirection}` at line {csv_line}"
+
+        assert -7 <= XDirection <= 7, \
+            f"X direction input has value `{XDirection}` at line {csv_line}"
```

Comment:
We (regrettably) shouldn't assume the CSV is 0-centered.

"\<direction\> direction input has inconsistent centering at line {csv_line}"
"\<direction\> direction input on line {csv_line} is out of range! Expected {range} - {range + 14}, got \`{\<direction\>}\`"

Reply:
In what situation would this not be the case? 
How do we know then how to parse the csv if this is not always the case? A parameter that specifies the format?

Reply:
There are still people who use 7-centered.

The most we can do is infer the center by checking unshared boundaries (> 7, < 0). If that fails, we may want to default to whichever number has the highest frequency, since neutral inputs are most likely the more common of the two. Alternatively, we can ask the user via input what center the script is.